### PR TITLE
sd-device: drop redundant slashes in syspath

### PR DESCRIPTION
**Pull request by yuwata**
C.f. https://bugzilla.redhat.com/show_bug.cgi?id=1787089

### CODE REVIEW

File: src/libsystemd/sd-device/sd-device.c [hunk 1]
```diff
@@ -152,6 +152,7 @@ int device_set_syspath(sd_device *device, const char *_syspath, bool verify) {
                 if (r < 0)
                         return log_debug_errno(r, "sd-device: Failed to get target of '%s': %m", _syspath);
 
+                path_simplify(syspath, false);
```
Comment (poettering) [inline src/libsystemd/sd-device/sd-device.c:155]: hmm, why did chase_symlinks() return a non-simplified path? that's surprising. I would assume we should always return a simplified path, hence simpliying the result of chase_symlinks() should never be necessary, and always be a NOP?

Comment (yuwata) [reply src/libsystemd/sd-device/sd-device.c:155]: I will check `chase_symlinks()`, but before this PR,
```
$ udevadm info --path ///devices/virtual/workqueue/.///writeback
P: ////devices/virtual/workqueue///writeback
L: 0
E: DEVPATH=////devices/virtual/workqueue///writeback
E: SUBSYSTEM=workqueue
```
and after this PR,
```
$ udevadm info --path ///devices/virtual/workqueue/.///writeback
P: /devices/virtual/workqueue/writeback
L: 0
E: DEVPATH=/devices/virtual/workqueue/writeback
E: SUBSYSTEM=workqueue
```
So, it seems `chase_symlinks()` simplifies only dots, not slashes.

Comment (poettering) [reply src/libsystemd/sd-device/sd-device.c:155]: we should fix that. this looks like a bug in chase_symlinks().

we should only return perfectly simplified paths in chase_symlinks() i am sure.

**Comment by yuwata** [comment]
Replaced by #14687.

**Pull request closed without merging**